apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -97,120 +128,196 @@ const LineDiagram = ({
     modalOpen: false
   });
 
+  const isBranchEnd = (index: number): boolean =>
```

Comment:
Would it be reasonable to define these functions outside the component function, so they are not defined as part of rendering the component?

Reply:
Even better - got rid of it!